### PR TITLE
Feature/stat tracker gv

### PR DESCRIPTION
**Pull request by GVarenhorst**
With this pull: 
1) Updated return values on tests after adding additional arguments
2) removed a pry to clean up code
3)added coach methods
4)added most and least accurate team methods with corresponding tests

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -181,18 +191,48 @@ def coach_stats
...
-  def worst_coach
-    worst_coach_stats = coach_stats.min_by {|coach, percentage| percentage}
+  def worst_coach(season)
+    worst_coach_stats = coach_stats(season).min_by {|coach, percentage| percentage}
     worst_coach_stats[0]
   end
 
-  def most_accurate_team
-    
+  def most_accurate_team(season)
```
Comment (DChavez18) [inline lib/stat_tracker.rb:211]: Good job on these methods!
I was thinking a parameter needed to be passed in like season!

**Comment by DChavez18** [comment]
I'm all for a merge here!

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -9,9 +10,10 @@ class StatTracker
   def initialize
     @games = []
     @teams = []
-    @season_hash = {}
+    @season_teams = {}
```
Comment (dani-wilson) [inline lib/stat_tracker.rb:13]: this is better for coder empathy! ;)

File: lib/stat_tracker.rb [hunk 3]
```diff
@@ -171,8 +180,9 @@ def lowest_scoring_home_team
     worst_team.team_name
   end
 
-  def coach_stats 
-    coach_data = @game_teams.group_by {|game| game.head_coach}
+  def coach_stats(season)
+    coach_data = find_season(season).game_teams.group_by {|game| game.head_coach}
+    # require 'pry'; binding.pry
```
Comment (algibson1) [inline lib/stat_tracker.rb:185]: We'll worry about it later when cleaning up our final code, but we'll need to remove all instances of pry

File: spec/stat_tracker_spec.rb [hunk 4]
```diff
@@ -84,30 +84,30 @@
   end
 
   it "has a best coach" do
-    expect(@stat_tracker.winningest_coach).to eq("Claude Julien")
+    expect(@stat_tracker.winningest_coach("20122013")).to eq("Claude Julien")
```
Comment (dani-wilson) [inline spec/stat_tracker_spec.rb:87]: This makes more sense. Nice

**Review by dani-wilson — APPROVED** [review]

File: lib/stat_tracker.rb [hunk 5]
```diff
@@ -181,18 +191,48 @@ def coach_stats
...
+
+  def least_accurate_team(season)
+    id_hash = find_season(season).game_teams.group_by { |game_team_object| game_team_object.team_id }
+    team_percentages = {}
+    id_hash.each do |team_id, games_array|
+      team_accuracy = games_array.map { |game_team_object| game_team_object.accuracy }
+      accurate_percentage = team_accuracy.sum.to_f / team_accuracy.length
+      team_percentages[team_id] = accurate_percentage
+    end
+    top_team_array = team_percentages.min_by {|team_id, percentage| percentage }
```
Comment (algibson1) [inline lib/stat_tracker.rb:233]: 212 through 218, and 226 through 232 are identical. We can put those in a helper method the same way we made a `coach_stats` helper and a `scoring_helper_method`

**Review by algibson1 — APPROVED** [review]

**Pull request merged**